src/backend/joanie/tests/test_api.py

```diff
@@ -0,0 +1,587 @@
+"""
+Test suite for API
+"""
+from datetime import datetime, timedelta
+
+from django.conf import settings
+from django.test import TestCase
+from django.test.utils import override_settings
+from django.utils import translation
+
+import arrow
+import jwt
+
+from joanie.core import enums, factories, models
+
+OPENEDX_COURSE_RUN_URI = "http://openedx.test/courses/course-v1:edx+%s/course"
+
+
+# pylint: disable=too-many-instance-attributes
+class APITestCase(TestCase):
+    """Test suite for API"""
+
+    def setUp(self):
+        """
+        Create a credible context to test API. Initialize some courses, course runs and products.
+        """
+        super().setUp()
+        translation.activate("en-us")
+        # First create a course product to learn how to become a good druid
+        # 1/ some course runs are required to became a good druid
+        self.bases_of_botany_run1 = factories.CourseRunFactory(
```

Comment:
I'm not a bit fan of factories in the setUp method because:
- it hides object creations making tests harder to read when they break. During a refactoring, you may have to fix hundreds of tests so it's much easier if test methods are self contained,
- you end up creating everything needed for all tests, slowing them down considerably. Creating just the minimum to prove your point for each test is in most cases much more efficient. In case all your tests need the exact same data that is hard to create (as seems to be your case here), a specific method is prefereable like you did for `_get_order_data` so you can make several, take arguments make things vary, and call it explicitly from your tests, making them easier to read...